packages/gator-permissions-snap/src/clients/priceApiClient.ts

```diff
@@ -44,6 +57,41 @@ export class PriceApiClient {
       );
     }
 
+    const { attempts = 1, delayMs = 1000 } = retryOptions ?? {};
+    let lastError: Error = new Error('No attempts made');
+
+    // Try up to initial attempt + retry attempts
+    for (let attempt = 0; attempt <= attempts; attempt++) {
+      try {
+        return await this.#fetchSpotPrice(caipAssetType, vsCurrency);
+      } catch (error) {
+        lastError = error as Error;
+
+        // Only retry on ResourceUnavailableError (5xx status codes)
+        if (error instanceof ResourceUnavailableError && attempt < attempts) {
```

Comment:
ResourceUnavailableError is actually (correctly) thrown on 5xx, 408, 429 in #fetchSpotPrice, but the comment indicates 5xx only.

Rather than catching in #fetchSpotPrice, should we rather return the response object / response + response code, and handle that directly in the retry loop, _then_ identify the appropriate error to throw? It feels like the responsibility of identifying a consumer ≈ errors is the responsibility of the public function, and the retry loop would be more logical with statuscodes rather than consumer friendly errors.